manifests/repo.pp

```diff
@@ -68,21 +68,18 @@
     'Debian': {
       include apt
 
+      apt::keyring { 'elastic.gpg':
+        source => $key_source,
+      }
+
       apt::source { 'elastic':
-        ensure   => 'present',
         comment  => $description,
         location => $base_url,
         release  => 'stable',
         repos    => 'main',
-        key      => {
-          'id'     => $key_id,
-          'source' => $key_source,
-        },
-        include  => {
-          'deb' => true,
-          'src' => false,
-        },
+        keyring  => '/etc/apt/keyrings/elastic.gpg',
```

Comment:
you could actually just use
```
key      => {
  'name'   => 'elastic.asc',
  'source' => $key_source,
}
``

Reply:
No, It will create apt::keyring when the hash contains 'name' :)
https://github.com/puppetlabs/puppetlabs-apt/blob/main/manifests/source.pp#L208